utils/utils.c

```diff
@@ -66,7 +66,6 @@ initGbConfig(void)
 
   gbConf->glfsLruCount = LRU_COUNT_DEF;
   gbConf->logLevel = GB_LOG_INFO;
-  snprintf(gbConf->logDir, PATH_MAX, "%s", GB_LOGDIR_DEF);
```

Comment:
Is there any reason for removing this line ?

Reply:
In both gluster-blockd and gluster-block they will call initLogging() immediately just after initGbConfig(), which will handle the whole logger stuff there, including the logDir's initialization, so this here in initGbConfig() make no sense and will cause the following issue:

[2019-05-28 15:01:58.991488] CRIT: trying to change logDir from /var/log/gluster-block to /var/log/gluster-block [at utils.c+495 :]

Reply:
I doubt I got you right here.

As we are initializing all the gbConf stuff, lets also keep LOGDIR defaults initialization here.

Further,
main()->initLogging()->initLogDirAndFiles(NULL)

```
static int                                                                       
initLogDirAndFiles(char *newLogDir)                                              
{                                                                                
...
  if (newLogDir) {                                                               
...
    LOG(dom, logLevel,                                                           
        "trying to change logDir from %s to %s", gbConf->logDir, logDir);        
  } else {                                                                 
...
```
If this the path, you are talking about, as newLogDir is NULL, this won't hit now.

in case, If you are taking about other route:
glusterBlockSetLogDir->initLogDirAndFiles(logDir)

We should change comparing paths with strcmp(logDir, gbConf->logDir) and adopt:
```
static bool is_same_path(const char* path1, const char* path2)                                                                                                                                                     
{                                                                                
  struct stat st1 = {0,};                                                        
  struct stat st2 = {0,};                                                        
                                                                                 
  if (!path1 || !path2)                                                          
    return false;                                                                
                                                                                 
  if (stat(path1, &st1) == -1 || stat(path2, &st2) == -1) {                      
    return false;                                                                
  }                                                                              
                                                                                 
  return st1.st_dev == st2.st_dev && st1.st_ino == st2.st_ino;                   
}  
``

Reply:
I am okay with this if you don't like this change :-)
I just want to collect all the logger init stuff into initLogging() routine because it will fallback to the GB_LOGDIR_DEF always if neither of the env and config option is set.

Reply:
@lxbsz I'm okay, moving this to initLogging(), as I see, other logfile initializing is happening there.

But please fix, if both the dirs are same it won't/shouldn't change again. And using strcmp for dirs is not the right thing, as users can give it dir/ or dir or dir//, all pointing to same path.

Reply:
Yeah, sure.
Fixing it now.
Thanks.